src/queue/mapping.rs

```diff
@@ -341,13 +341,13 @@ impl Mapping {
     /// Bulk dequeue operation. Returns number of entries succesfully dequeued
     pub fn dequeue_bulk_bytes<'a>(
         &mut self,
-        data: &mut [(Option<&'a mut String>, &'a mut Vec<u8>)],
+        data: &mut Vec<(String, Vec<u8>)>,
         copy_topic: bool,
     ) -> usize {
         let buffer_pool = unsafe { &mut *self.buffer_pool.get() };
         // trim data to the smaller of the two (pending or data length)
         let dequeued = data.len().min(self.pending());
-        let data = &mut data[..dequeued];
+        data.truncate(dequeued);
```

Comment:
What if we skipped truncating the vec here and let the caller deal with only reading up to the returned count? It would allow us to not pass a vec here and instead just pass a mut reference to a slice